layers/core_validation.cpp

```
@@ -11098,138 +11098,137 @@ bool CoreChecks::ValidateCreateSwapchain(const char *func_name, VkSwapchainCreat
                     "%s: surface capabilities not retrieved for this physical device", func_name))
             return true;
-    } else {  // have valid capabilities
-        auto &capabilities = physical_device_state->surfaceCapabilities;
-        // Validate pCreateInfo->minImageCount against VkSurfaceCapabilitiesKHR::{min|max}ImageCount:
-        if (pCreateInfo->minImageCount < capabilities.minImageCount) {
-            if (log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT, HandleToUint64(device),
-                        "VUID-VkSwapchainCreateInfoKHR-minImageCount-01271",
-                        "%s called with minImageCount = %d, which is outside the bounds returned by "
-                        "vkGetPhysicalDeviceSurfaceCapabilitiesKHR() (i.e. minImageCount = %d, maxImageCount = %d).",
-                        func_name, pCreateInfo->minImageCount, capabilities.minImageCount, capabilities.maxImageCount))
-                return true;
-        }
+    }
 
-        if ((capabilities.maxImageCount > 0) && (pCreateInfo->minImageCount > capabilities.maxImageCount)) {
-            if (log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT, HandleToUint64(device),
-                        "VUID-VkSwapchainCreateInfoKHR-minImageCount-01272",
-                        "%s called with minImageCount = %d, which is outside the bounds returned by "
-                        "vkGetPhysicalDeviceSurfaceCapabilitiesKHR() (i.e. minImageCount = %d, maxImageCount = %d).",
-                        func_name, pCreateInfo->minImageCount, capabilities.minImageCount, capabilities.maxImageCount))
-                return true;
-        }
+    VkSurfaceCapabilitiesKHR *capabilities = &physical_device_state->surfaceCapabilities;
+    DispatchGetPhysicalDeviceSurfaceCapabilitiesKHR(physical_device_state->phys_device, pCreateInfo->surface, capabilities);
```

This is the gist of the change -- making the query again and updating the local state.  The rest of the changes are mostly indent diffs.

---

Should we be saving state in PreCallValidate? Is there any reason to store these if we are going to get a fresh copy each time?

---

Should the other vkGetPhysicalDeviceSurface*State validation dispatch their own queries too?  By changing those UNCALLED errors to warnings we could be masking an actual error if someone uses an unsupported format without querying first.